ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/committer/transaction/validation/CommitterModelConformanceValidator.scala

```diff
@@ -108,19 +112,21 @@ private[transaction] class ModelConformanceValidator(engine: Engine, metrics: Me
       } yield ()
       stepResult.fold(identity, _ => StepContinue(transactionEntry))
     } catch {
-      case missingInputErr @ Err.MissingInputState(key) =>
+      case missingInputErr @ Err.MissingInputState(
+            key
+          ) => // Missing package or contract not specified as input
         logger.error(
```

Comment:
```suggestion
        logger.warn(
```

Reply:
Oh, my apologies; I thought this could also be triggered by pruned contracts. Ignore me.

Reply:
I’m confused, why can this not be triggered by pruning?

Reply:
In this logic, it looks to me like this can be thrown only [here](https://github.com/digital-asset/daml/blob/main/ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/committer/CommitContext.scala#L56), i.e. if the participant doesn't declare a needed input. I don't see how this can be caused by committer pruning but maybe I am missing something, am I?

Reply:
This is not something we were able to reproduce. The error we get from PKV reference architecture conformance tests:
```
30-11-2021 11:26:00.624 UTC|ERROR||c.d.l.p.s.k.KeyValueConsumption|in-memory-ledger-akka.actor.default-dispatcher-16|||||VALIDATION_FAILURE(4,RWArchiv): Validation failure: Contract could not be found with id 00e471ec59434402166e631f1a54e84f01db9073ad38c8c32c4d8ba4e1f4789a44|{participantId: "conformance-test", offset: "01000000000006ea0000000000000000", err-context: "{location=TransactionRejections.scala:544}"}|
```
comes from here: https://github.com/digital-asset/daml/pull/12108/files#diff-7cde65905babc148f1f6872cdc5c034ae66d0390e09d4f764c47910ae8b5d2d8R108

This PR seems to be sufficient to fix this particular issue.

Reply:
Sorry @fabiotudone-da, didn't refresh the page and missed your comment

Reply:
@hubert-da I forgot to add that it indeed does, already tested with a local SDK build.

Reply:
@hubert-da It shouldn't be in this particular case, as [that](https://github.com/digital-asset/daml/blob/main/ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/committer/CommitContext.scala#L56) is a validation on the declared transaction inputs.

Reply:
Ok, makes sense, thanks.

Reply:
thanks for the explanation, sounds good